packages/nimbus/src/components/primary-action-dropdown/primary-action-dropdown.tsx

```diff
@@ -0,0 +1,347 @@
+
+  /**
+   * Find a specific Menu.Item by ID to populate the primary button content in split mode.
+   * This searches through Menu.Items (including those nested in Menu.Sections).
+   *
+   * @param children - The menu children to search through
+   * @param targetId - The ID of the Menu.Item we want to find
+   * @returns The Menu.Item content and properties, or null if not found
+   */
+  const findMenuItemById = (
+    children: React.ReactNode,
+    targetId: string
+  ): {
+    content: React.ReactNode;
+    isDisabled: boolean;
+    isCritical: boolean;
+  } | null => {
+    let found: {
+      content: React.ReactNode;
+      isDisabled: boolean;
+      isCritical: boolean;
+    } | null = null;
+
+    React.Children.forEach(children, (child) => {
+      if (found) return;
+
+      // Direct Menu.Item match
+      if (isMenuItemWithId(child) && child.props.id === targetId) {
+        found = {
+          content: child.props.children,
+          isDisabled: child.props.isDisabled || false,
+          isCritical: child.props.isCritical || false,
+        };
+        return;
+      }
+
+      // Recurse into Menu.Section or any component with children
+      if (hasChildren(child)) {
+        found = findMenuItemById(child.props.children, targetId);
+      }
+    });
+
+    return found;
+  };
+
+  /**
+   * Check if there are any actionable (enabled) Menu.Items in the children.
+   * This is used to determine if the dropdown trigger should be disabled.
+   */
+  const hasActionableMenuItems = (children: React.ReactNode): boolean => {
+    let hasActionable = false;
+
+    React.Children.forEach(children, (child) => {
+      if (hasActionable) return;
+
+      // Check if this is an enabled Menu.Item
+      if (isMenuItemWithId(child) && !child.props.isDisabled) {
+        hasActionable = true;
+        return;
+      }
+
+      // Recurse into Menu.Section or any component with children
+      if (hasChildren(child)) {
+        hasActionable = hasActionableMenuItems(child.props.children);
+      }
+    });
+
+    return hasActionable;
+  };
+
+  /**
+   * Get the primary button content for split mode.
+   * Uses defaultAction if provided, otherwise shows standard fallback text.
+   * This is the main function that resolves "what should the primary button show?".
+   */
+  const getPrimaryButtonContent = (): {
+    content: React.ReactNode;
+    isDisabled: boolean;
+    actionId: string | null;
+  } => {
+    // If defaultAction is specified, try to use it directly
+    if (defaultAction) {
+      const item = findMenuItemById(menuItems, defaultAction);
+      if (item) {
+        return {
+          content: item.content,
+          isDisabled: item.isDisabled || false,
+          actionId: defaultAction,
+        };
+      }
+    }
+
+    // If no defaultAction found or no defaultAction specified, show standard fallback
+    return {
+      // TODO: localize this
```

Comment:
We'll have to localize a couple things, I don't expect consumers to have to specify fallbacks. A later time